watchtower/wtclient/client.go

```diff
@@ -561,6 +561,7 @@ func (c *TowerClient) backupDispatcher() {
 			// Wait until we receive the newly negotiated session.
 			// All backups sent in the meantime are queued in the
 			// revoke queue, as we cannot process them.
+		awaitSession:
```

Comment:
It's possible for the client to not gracefully shut down if it's waiting here. In the `select` statement for once we have an active session we seem to handle the case where the backup queue is shut down, but not here.

Relevant profile:
```
1 @ 0x1030e1f 0x10407f8 0x19b991d 0x105e331
#	0x19b991c	github.com/lightningnetwork/lnd/watchtower/wtclient.(*TowerClient).backupDispatcher+0xc5c	/Users/user/git/lnd/watchtower/wtclient/client.go:611

1 @ 0x1030e1f 0x10407f8 0x19bc7d5 0x105e331
#	0x19bc7d4	github.com/lightningnetwork/lnd/watchtower/wtclient.(*sessionNegotiator).negotiationDispatcher+0x134	/Users/user/git/lnd/watchtower/wtclient/session_negotiator.go:177

1 @ 0x1030e1f 0x10407f8 0x19bd102 0x105e331
#	0x19bd101	github.com/lightningnetwork/lnd/watchtower/wtclient.(*sessionNegotiator).negotiate+0x6a1	/Users/user/git/lnd/watchtower/wtclient/session_negotiator.go:236

1 @ 0x1030e1f 0x10412d9 0x10412af 0x1040f59 0x10751f5 0x19c2bc1 0x10742f3 0x19b8484 0x1aa9245 0x10742f3 0x1a85f34 0x1a4b702 0x1ab6b06 0x1030a2c 0x105e331
#	0x1040f58	sync.runtime_Semacquire+0x38								/usr/local/go/src/runtime/sema.go:56
#	0x10751f4	sync.(*WaitGroup).Wait+0x64								/usr/local/go/src/sync/waitgroup.go:130
#	0x19c2bc0	github.com/lightningnetwork/lnd/watchtower/wtclient.(*TowerClient).Stop.func1+0xa0	/Users/user/git/lnd/watchtower/wtclient/client.go:415
#	0x10742f2	sync.(*Once).Do+0xb2									/usr/local/go/src/sync/once.go:44
#	0x19b8483	github.com/lightningnetwork/lnd/watchtower/wtclient.(*TowerClient).Stop+0x53		/Users/user/git/lnd/watchtower/wtclient/client.go:387
#	0x1aa9244	github.com/lightningnetwork/lnd.(*server).Stop.func1+0x354				/Users/user/git/lnd/server.go:1312
#	0x10742f2	sync.(*Once).Do+0xb2									/usr/local/go/src/sync/once.go:44
#	0x1a85f33	github.com/lightningnetwork/lnd.(*server).Stop+0x53					/Users/user/git/lnd/server.go:1283
#	0x1a4b701	github.com/lightningnetwork/lnd.Main+0x1241						/Users/user/git/lnd/lnd.go:489
#	0x1ab6b05	main.main+0x25										/Users/user/git/lnd/cmd/lnd/main.go:14
#	0x1030a2b	runtime.main+0x20b									/usr/local/go/src/runtime/proc.go:200
```

Reply:
yep this fixed locally setting a `ForceQuitDelay` in the client's conf

Reply:
Why should it still wait for the delay if there aren't any queued tasks though?